apicapi/apic_manager.py

```diff
@@ -218,12 +218,15 @@ def ensure_infra_created_on_apic(self):
         setup profiles for switches, modules and ports
         """
 
+        for domain in self.domains:
+            if (self.provision_infra or
+                getattr(domain, 'vmm_type', None) ==
+                    apic_domain.APIC_VMM_TYPE_OPENSTACK):
```

Comment:
actually you can also call domain.create() even if the vmm_type == APIC_VMM_TYPE_VMWARE. This is because for vmware, user has to create the vmm domain beforehand and we check that vmm domain does exist inside domain.create() call now. So even if self.provision_infra is false, we should still check that for the vmware vmm_type.